File: rximagepicker_support_zhihu/src/main/java/com/qingmei2/rximagepicker_extension_zhihu/ui/ZhihuImagePickerActivity.java
package com.qingmei2.rximagepicker_extension_zhihu.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.qingmei2.rximagepicker.entity.Result;
import com.qingmei2.rximagepicker.ui.ActivityPickerViewController;
import com.qingmei2.rximagepicker_extension.entity.SelectionSpec;
import com.qingmei2.rximagepicker_extension_zhihu.R;

import io.reactivex.functions.Action;
import io.reactivex.functions.Consumer;

public class ZhihuImagePickerActivity extends AppCompatActivity {

    private ZhihuImagePickerFragment fragment;

    public static final int REQUEST_CODE_PREVIEW = 23;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        setTheme(SelectionSpec.Companion.getInstance().getThemeId());
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_picker_zhihu);
        displayPickerView();
    }

    private void displayPickerView() {
        fragment = new ZhihuImagePickerFragment();
        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.fl_container, fragment)
                .commit();

        fragment.pickImage()
                .subscribe(new Consumer<Result>() {
                    @Override
                    public void accept(Result result) throws Exception {
                        ActivityPickerViewController.getInstance().emitResult(result);
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        ActivityPickerViewController.getInstance().emitError(throwable);
                    }
                }, new Action() {
                    @Override
                    public void run() throws Exception {
                        closure();
                    }
                });
    }

    public void closure() {
        ActivityPickerViewController.getInstance().endResultEmitAndReset();
        finish();
    }

    @Override
    public void onBackPressed() {
        closure();
    }

}

File: rximagepicker_support_zhihu/src/main/java/com/qingmei2/rximagepicker_extension_zhihu/ui/ZhihuImagePickerActivity.kt
package com.qingmei2.rximagepicker_extension_zhihu.ui

import android.Manifest
import android.annotation.SuppressLint
import android.content.pm.PackageManager
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import com.qingmei2.rximagepicker.entity.Result
import com.qingmei2.rximagepicker.ui.ActivityPickerViewController
import com.qingmei2.rximagepicker.ui.camera.BasicCameraFragment
import com.qingmei2.rximagepicker_extension.entity.SelectionSpec
import com.qingmei2.rximagepicker_extension.ui.adapter.AlbumMediaAdapter
import com.qingmei2.rximagepicker_extension_zhihu.R
import io.reactivex.Observer
import io.reactivex.disposables.Disposable

class ZhihuImagePickerActivity : AppCompatActivity(), AlbumMediaAdapter.OnPhotoCapture {

    private val fragment: ZhihuImagePickerFragment = ZhihuImagePickerFragment()

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(SelectionSpec.instance.themeId)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_picker_zhihu)

        requestPermissionAndDisplayGallery()
    }

    private fun requestPermissionAndDisplayGallery() {
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this,
                    arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), 99)
        } else {
            displayPickerView()
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
            displayPickerView()
        } else {
            closure()
        }
    }

    @SuppressLint("CheckResult")
    private fun displayPickerView() {
        supportFragmentManager
                .beginTransaction()
                .add(R.id.fl_container, fragment)
                .commitAllowingStateLoss()

        fragment.pickImage().subscribe(observer)
    }

    override fun capture() {
        BasicCameraFragment()
                .apply {
                    display(
                            fragmentActivity = this@ZhihuImagePickerActivity,
                            viewContainer = R.id.fl_container,
                            configuration = null
                    )
                    pickImage().subscribe(observer)
                }
    }

    private val observer = object : Observer<Result> {

        override fun onComplete() = closure()

        override fun onSubscribe(p0: Disposable) {
            // do nothing
        }

        override fun onNext(result: Result) =
                ActivityPickerViewController.instance.emitResult(result = result)

        override fun onError(e: Throwable) =
                ActivityPickerViewController.instance.emitError(e)

    }

    fun closure() {
        ActivityPickerViewController.instance.endResultEmitAndReset()
        finish()
    }

    override fun onBackPressed() {
        closure()
    }

    companion object {

        const val REQUEST_CODE_PREVIEW = 23
    }

}
